### PR TITLE
[FEA] Improve distributed api docs

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
Improves API docs for the distributed API. 

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Review by dallasfoster — APPROVED** [review]
Small comments, seems good to me otherwise.

File: docs/api/modulus.distributed.rst [hunk 1]
```diff
@@ -1,15 +1,169 @@
...
+How does this work?
+"""""""""""""""""""
+
+An important aspect of the ``DistributedManager`` is that it is follows the 
+`Borg pattern <https://code.activestate.com/recipes/66531-singleton-we-dont-need-no-stinkin-singleton-the-bo/>`_.
+This means that ``DistributedManager`` essentially functions like a singleton 
+class and once configured, all utilities in Modulus can access the same configuration 
+and adapt to the specified distributed structure.
+
+For example, see the consturctor of the ``DistributedAFNO`` class:
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/modulus.distributed.rst]: Thanks Dallas, fixed all the typos!

File: docs/api/modulus.distributed.rst [hunk 2]
```diff
@@ -1,15 +1,169 @@
 Modulus Distributed
 ===================
 
+.. automodule:: modulus.distributed
+.. currentmodule:: modulus.distributed
+
+Distributed utilites in Modulus are designed to simplify implementation of parallel training and
+inference scripts easier by providing a unified way to configure and query parameters associated
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/modulus.distributed.rst]: Fixed

**Review by NickGeneva — APPROVED** [review]

**Pull request merged**